node/full_node_test.go

```diff
@@ -9,52 +9,37 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/mock"
-	"github.com/stretchr/testify/require"
 
 	abci "github.com/cometbft/cometbft/abci/types"
-	"github.com/cometbft/cometbft/libs/log"
-	proxy "github.com/cometbft/cometbft/proxy"
-	"github.com/cometbft/cometbft/types"
 	"github.com/libp2p/go-libp2p/core/crypto"
 	"github.com/libp2p/go-libp2p/core/peer"
 
 	testutils "github.com/celestiaorg/utils/test"
 
-	"github.com/rollkit/rollkit/config"
 	"github.com/rollkit/rollkit/mempool"
 	"github.com/rollkit/rollkit/test/mocks"
 )
 
 // TestStartup checks if the node starts and stops without any errors
 func TestStartup(t *testing.T) {
-	assert := assert.New(t)
-	require := require.New(t)
-
 	ctx := context.Background()
-	node := initializeAndStartNode(ctx, require, assert)
-	defer cleanUpNode(node, assert)
+	node := initializeAndStartFullNode(ctx, t)
+	defer cleanUpNode(node, t)
 }
 
 func TestMempoolDirectly(t *testing.T) {
-	assert := assert.New(t)
-	require := require.New(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
Is there a reason why `defer cancel()` is called here but not in `TestStartup`?

Reply:
I removed some of the `defer cancel()` since they are not necessary when using `cleanUpNode`.  If we are using `initializeAndStartNode` with `cleanUpNode` we can use create the background context since `NewNode` creates the cancel function.